src/components/MapView/Layers/BoundaryLayer/index.tsx

```diff
@@ -28,32 +38,77 @@ function BoundaryLayer({ layer }: { layer: BoundaryLayerProps }) {
 
   const isPrimaryLayer = isPrimaryBoundaryLayer(layer);
 
-  const onClickFunc = (evt: any) => {
+  const onHoverHandler = (evt: any) => {
+    dispatch(hidePopup());
+    onToggleHover('pointer', evt.target);
     const coordinates = evt.lngLat;
     const locationName = getFullLocationName(
       layer.adminLevelNames,
       evt.features[0],
     );
+
     const locationLocalName = getFullLocationName(
       layer.adminLevelLocalNames,
       evt.features[0],
     );
+
     dispatch(showPopup({ coordinates, locationName, locationLocalName }));
+  };
+
+  const onClickFunc = (evt: any) => {
+    const { properties } = evt.features[0];
+
     // send the selection to the map selection layer. No-op if selection mode isn't on.
     dispatch(
       toggleSelectedBoundary(evt.features[0].properties[layer.adminCode]),
     );
+
+    const selectedLayerWMS: undefined | WMSLayerProps = selectedLayers.find(
+      l => l.type === 'wms' && l.chartData,
+    ) as WMSLayerProps;
+
+    if (!selectedLayerWMS) {
+      return;
+    }
+
+    const { serverLayerName, title, chartData } = selectedLayerWMS;
+
+    const { levels, url, type: chartType } = chartData!;
+
+    const lowestLevelId = levels[levels.length - 1].id;
+
+    const boundaryProps = levels.reduce(
+      (obj, item) => ({
+        ...obj,
+        [item.id]: {
+          code: properties[item.id],
+          urlPath: item.path,
+          name: properties[item.name],
+        },
+      }),
+      {},
+    );
+
+    const adminBoundaryParams: AdminBoundaryParams = {
+      title,
+      boundaryProps,
+      serverParams: { layerName: serverLayerName, url },
+      id: lowestLevelId,
+      chartType,
+    };
+
+    dispatch(setBoundaryParams(adminBoundaryParams));
   };
 
   // Only use mouse effects and click effects on the main layer.
-  const { fillOnMouseEnter, fillOnMouseLeave, fillOnClick } = isPrimaryLayer
+  const { fillOnMouseMove, fillOnMouseLeave, fillOnClick } = isPrimaryLayer
     ? {
-        fillOnMouseEnter: (evt: any) => onToggleHover('pointer', evt.target),
+        fillOnMouseMove: onHoverHandler,
         fillOnMouseLeave: (evt: any) => onToggleHover('', evt.target),
         fillOnClick: onClickFunc,
       }
     : {
-        fillOnMouseEnter: undefined,
+        fillOnMouseMove: undefined,
```

Comment:
I am not a fan of changing this behavior to onHover. In addition, this breaks how we display the results of analysis layers in popups